spec/jobs/payment_provider_customers/gocardless_checkout_url_job_spec.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe PaymentProviderCustomers::GocardlessCheckoutUrlJob, type: :job do
+  subject { described_class }
+
+  let(:gocardless_customer) { create(:gocardless_customer) }
+
+  let(:gocardless_service) { instance_double(PaymentProviderCustomers::GocardlessService) }
+
+  it 'calls generate_checkout_url method' do
+    allow(PaymentProviderCustomers::GocardlessService).to receive(:new)
+      .with(gocardless_customer)
+      .and_return(gocardless_service)
+    allow(gocardless_service).to receive(:generate_checkout_url)
+      .and_return(BaseService::Result.new)
+
+    subject.perform_now(gocardless_customer)
```

Comment:
RSpec/NamedSubject: Name your test subject if you need to reference it explicitly. (https://rspec.rubystyle.guide/#use-subject, https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/NamedSubject)